### PR TITLE
A pack for the Tesla Automobile(s)

### PR DESCRIPTION
**Pull request by tonybaloney**
### Status 

A pack for your garage of Tesla cars (Tesla not included)

- new pack, complete

### Description

Includes actions for : 

* `charge_vehicle` - Set the charge pattern to standard
* `charge_vehicle_max_range` - Set the charge pattern to max
* `flash_lights` - Flash the lights of the car
* `get_vehicle` - Get the details of a vehicle
* `honk_horn` - Honk the horn
* `list_vehicles` - List vehicles in your garage
* `lock_vehicle` - Lock the car
* `open_charge_port_door` - Open the charge port hatch
* `set_vehicle_sun_roof` - Set the position of the sun roof
* `set_vehicle_temperature` - Set the temperature of the car
* `start_vehicle_ac` - Start the AC
* `start_vehicle_charge` - Start the charging process
* `stop_vehicle_ac` - Stop the AC
* `stop_vehicle_charge` - Stop the charging process
* `unlock_vehicle` - Unlock the car
* `wake_vehicle` - Wake the vehicle up

### Checklist (tick everything that applies)

- [x] Metadata: pack.yaml, icon, structure (required for new packs)
- [x] Version bump (required for changed packs)
- [x] Code linting (required, can be done after the PR checks)
- [x] Tests (not required but really recommended)

### CODE REVIEW

**Comment by manasdk** [comment]
@tonybaloney how does one test such a pack?

**Comment by tonybaloney** [comment]
@manasdk don't you have a tesla(s)?

**Comment by manasdk** [comment]
> @manasdk don't you have a tesla(s)?

I don't even have a 32:1 scale model let alone a real one :(

**Comment by emedvedev** [comment]
There's a weird-looking grey border on the icon, and icon size should be different.

You're an interesting man, Anthony.

**Comment by emedvedev** [comment]
Since we can't test the pack, you'll need to take a video of you and one of your many Teslas in frame. You giving StackStorm commands and the Tesla responding to them should be clearly visible. Please also include your home address in pack metadata so we could direct an angry mob in case something goes wrong.

File: packs/tesla/aliases/honk_horn.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+name: "tesla.honk_horn"
+action_ref: "tesla.honk_horn"
+description: "Honk your horn"
+formats:
+    - "honk {{vin}}"
+    - "beep-beep {{vin}}"
+    - "meep-meep {{vin}}"
+    - "honk-honk {{vin}}"
+result:
+    format: "**Honk Honk**"
```
Comment (emedvedev) [inline packs/tesla/aliases/honk_horn.yaml]: Please consider using `{~}` to output the result in plaintext. It's usually recommended for short aliases.

File: packs/tesla/aliases/honk_horn.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+name: "tesla.honk_horn"
+action_ref: "tesla.honk_horn"
+description: "Honk your horn"
+formats:
+    - "honk {{vin}}"
+    - "beep-beep {{vin}}"
+    - "meep-meep {{vin}}"
+    - "honk-honk {{vin}}"
```
Comment (emedvedev) [inline packs/tesla/aliases/honk_horn.yaml]: Consider adding optional exclamation marks at the end (one or more).

**Comment by tonybaloney** [comment]
also, FWIW, any self-respecting IFTTT/IOT software HAS to have a Tesla integration. I don't know why, it just does. @emedvedev @manasdk

**Comment by tonybaloney** [comment]
I will investigate having Tesla loan me an automobile for testing purposes. Or ask @lakshmi-kannan

**Comment by armab** [comment]
```py
for vehicle in list_vehicles:
    honk_horn(vehicle)
```

:car: :postal_horn:

**Comment by Kami** [comment]
Nice, maybe we can expense Tesla rental now to test this :P

**Comment by emedvedev** [comment]
I called Elon, he said this pack is ok. Merging.

**Pull request merged**